SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -2371,6 +2363,20 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
 
 		QueryHandler.ProcessOps(Connection->GetWorkerMessages());
 	}
+
+	// Broadcast network failure if any network errors occurred
+	// NOTE: this should be performed at the end of this function to prevent network disconnection errors
+	if (!NetworkFailureError.IsEmpty())
+	{
+		if (USpatialGameInstance* GameInstance = GetGameInstance())
+		{
+			if (GEngine != nullptr && GameInstance->GetWorld() != nullptr)
+			{
+				GEngine->BroadcastNetworkFailure(GameInstance->GetWorld(), this, ENetworkFailure::OutdatedClient, NetworkFailureError);
+			}
+		}
+		NetworkFailureError.Empty();
+	}
```

Comment:
```suggestion
	// Broadcast network failure if any network errors occurred
	// NOTE: this should be performed at the end of this function to prevent network disconnection errors
	if (PendingNetworkFailure)
	{
		if (USpatialGameInstance* GameInstance = GetGameInstance())
		{
			if (GEngine != nullptr && GameInstance->GetWorld() != nullptr)
			{
				GEngine->BroadcastNetworkFailure(GameInstance->GetWorld(), this, PendingNetworkFailure->FailureType, PendingNetworkFailure->Message);
			}
		}
		NetworkFailureError.Reset();
	}
```